Explain diversity of approaches (#51)
intarchboard/draft-obliviousness: draft-iab-privacy-partitioning.md

requires separating contexts, such as by using proxying to conceal a client IP address
that would otherwise be used as an identifier.

## Approaches to Partitioning

> While all of the partitioning protocols described in this document use the
> basic techniques of encryption and connection separation, each one has a

Reviewer: ```suggestion
While all of the partitioning protocols described in this document separate contexts
using cryptography, connection separation, or a combination of the two, each one has a
```

Because DAP doesn't need connection separation -- it just uses secret sharing.

Author: ```suggestion
While all of the partitioning protocols described in this document create
separate contexts using encryption and/or connection separation, each one has a
```

unique approach that results in different sets of contexts. Since many of
these protocols are new, it is yet to be seen how each approach will be
used at scale across the Internet, and what new models will emerge in the
future.

There are multiple factors that lead to a diversity in approaches to

> partitioning:

Reviewer: ```suggestion
partitioning, including:
```

To make more clear that this list is not exhaustive.

- Adding privacy partitioning to existing protocol ecosystems places
requirements and constraints on how contexts are constructed. CONNECT-style
proxying is intended to work with servers that are unaware of privacy contexts,
requiring more intermediaries to provide strong separation guarantees.
Oblivious HTTP, on the other hand, assumes servers that cooperate with context
separation, and thus reduces the overall number of elements in the solution.

- Whether or not information exchange needs to happen bidirectionally in an
interactive fashion determines how contexts can be separated. Some use cases,
like metrics collection for PPM, can occur with information flowing only from
clients to servers, and can function even when clients are no longer connected.
Privacy Pass is an example of a case that can be either interactive or not,

> depending on if tokens can be cached and reused. CONNECT-style proxying and
> Oblivious HTTP require bidirectional and interactive communication.

Reviewer: ```suggestion
depending on if tokens can be cached and reused. CONNECT-style proxying and
Oblivious HTTP often require bidirectional and interactive communication.
```

I wouldn't say that OHTTP requires bidirectional communication since apps could implement fire-and-forget (clients POST and then don't care about the response).

> 

Reviewer: ```suggestion
- The threat model, and in particular whether or not the servers providing support for context
separation are trusted. The degree to which contexts need to be partitioned depends on part
on how much clients trust them with such information. For example, in Oblivious HTTP, clients
trust relays to learn that clients are accessing a particular application-specific gateway. If clients
do not trust relays with this option, they may opt for a multi-hop CONNECT-style proxy approach
wherein no single party learns whether specific clients are accessing a specific application. Indeed,
this is the default trust model for systems like Tor, where multiple hops are used to drive down the
probability of privacy violations due to collusion or related attacks.
```

Reviewer 2: This first sentence seem broken, so maybe I don't get the point. However, I think this is all pretty much the core of what this document says but I would put is the other way a round - in order to minimise the needed trust, you should partition your data as much as you can - however this means of course also if you can't partition certain data because that's needed for the functionality you are requesting, then you might need a higher level of trust the more data you share (or maybe it's not so much about the "amount" of data but more particularly about the kind of data).

Reviewer: This suggested point is that the trust model dictates how you partition, which is not a factor we explicitly cover in either of the other two points, which are mostly about compatibility with existing systems and usability.

Author: Yeah I think this is a good point to add, but I think we should make the first sentence complete.

Author: ```suggestion
- The degree to which contexts need to be partitioned depends in part
on the client's threat models and level of trust in various protocol participants. For example,
in Oblivious HTTP, clients allow relays to learn that clients are accessing a particular
application-specific gateway. If clients do not trust relays with this information, they can
instead use a multi-hop CONNECT-style proxy approach wherein no single party learns
whether specific clients are accessing a specific application. This is the default trust model
for systems like Tor, where multiple hops are used to drive down the probability of privacy
violations due to collusion or related attacks.
```
